src/formatDate.js

```diff
@@ -51,6 +51,43 @@
 
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const newDate = [];
+  const oldDate = date.split(fromFormat[fromFormat.length - 1]);
+
+  function findSameElement(index) {
+    for (const oldElement of fromFormat) {
+      const indexOfOld = fromFormat.indexOf(oldElement);
+      let newElement = oldDate[indexOfOld];
+
+      if (oldElement === toFormat[index]) {
+        newDate.push(newElement);
+        break;
+      }
+
+      const isBothYears = oldElement.includes('Y')
+      && toFormat[index].includes('Y');
+      const isNewLengthBigger = oldElement.length < toFormat[index].length;
+
+      if (isBothYears && !isNewLengthBigger) {
+        newElement = oldDate[indexOfOld].slice(2);
+        newDate.push(newElement);
+      }
+
+      if (isBothYears && isNewLengthBigger && (+oldDate[indexOfOld] < 30)) {
+        newElement = 20 + oldDate[indexOfOld];
+        newDate.push(newElement);
+      } else if (isBothYears && isNewLengthBigger) {
+        newElement = 19 + oldDate[indexOfOld];
+        newDate.push(newElement);
+      }
+    }
+  }
+
+  findSameElement(0);
+  findSameElement(1);
+  findSameElement(2);
```

Comment:
the code repeats three times from 0 to 2,
you can use loop instead of a function